app/src/main/java/com/github/se/icebreakrr/ui/navigation/NavigationAction.kt

```diff
@@ -0,0 +1,51 @@
+package com.github.se.icebreakrr.ui.navigation
+
+import androidx.navigation.NavGraph.Companion.findStartDestination
+import androidx.navigation.NavHostController
+
+open class NavigationActions(
+    private val navController: NavHostController,
+) {
+  /**
+   * Navigate to the specified [TopLevelDestination]
+   *
+   * @param destination The top level destination to navigate to Clear the back stack when
+   *   navigating to a new destination This is useful when navigating to a new screen from the
+   *   bottom navigation bar as we don't want to keep the previous screen in the back stack
+   */
+  open fun navigateTo(destination: TopLevelDestination) {
+    navController.navigate(destination.route) {
+      popUpTo(navController.graph.findStartDestination().id) {
+        saveState = true
+        inclusive = false // Set to false so as not to delete the beginning
+      }
+      launchSingleTop = true
+      if (destination.route != Route.AUTH) {
+        restoreState = true
+      }
+    }
+  }
+
+  /**
+   * Navigate to the specified screen.
+   *
+   * @param screen The screen to navigate to
+   */
+  open fun navigateTo(screen: String) {
+    navController.navigate(screen)
+  }
+
+  /** Navigate back to the previous screen. */
+  open fun goBack() {
+    navController.popBackStack()
+  }
+
+  /**
+   * Get the current route of the navigation controller.
+   *
+   * @return The current route
+   */
+  open fun currentRoute(): String {
+    return navController.currentDestination?.route ?: "unknown"
```

Comment:
If the current route is null (which should not happen), we can redirect the user to the auth screen. If the current route is "unknown," the application will likely crash.

Reply:
Yes, I initially planned to handle the error elsewhere, but it's better to address it here